url79d4f7c7ed_schanq_Live_Algorithm_tgz-pJ8-algorithms_and_systems_AI_Evolutionary_GeneticAlgorithmJ8/src/expert_system/Rhythmic_Block.java

```diff
@@ -0,0 +1,805 @@
+	SEMI_QUAVERS4_FIRST_AND_SECOND_OFF(1),
+	SEMI_QUAVERS4_SECOND_AND_THIRD_OFF(1),
+	SEMI_QUAVERS4_THIRD_AND_FORTH_OFF(1),
+	SEMI_QUAVERS4_FIRST_AND_THIRD_OFF(1),
+	SEMI_QUAVERS4_SECOND_AND_FORTH_OFF(1),
+	SEMI_QUAVERS8(2),
+	SEMI_QUAVERS8_FIRST_THIRD_FIFTH_AND_SEVENTH_OFF(2),
+	SEMI_QUAVERS8_SECOND_FORTH_SIXTH_AND_EIGTH_OFF(2),
+	SEMI_QUAVERS8_SECOND_THIRD_FIFTH_AND_SIXTH_OFF(2),
+	SEMI_QUAVERS8_FIRST_SECOND_FORTH_AND_FIFTH_OFF(2),
+	SEMI_QUAVERS8_SECOND_FIFTH_SEVENTH_AND_EIGTH_OFF(2),
+	SEMI_QUAVERS8_THIRD_SIXTH_AND_EIGTH_OFF(2),
+	SEMI_QUAVERS_SWUNG_T1(1),
+	SEMI_QUAVERS_SWUNG_T2(1),
+	SEMI_QUAVERS_SWUNG_T3(1),
+	SEMI_QUAVERS_SWUNG_T4(1),
+	TRIPLET_SEMI_QUAVERS_FULL(1),
+	TRIPLET_SEMI_QUAVERS_VAR1(1),
+	TRIPLET_SEMI_QUAVERS_VAR2(1),
+	TRIPLET_SEMI_QUAVERS_VAR3(1),
+	DEMI_SEMI_QUAVERS8(1),
+	DEMI_SEMI_QUAVERS8_FIRST_THIRD_FIFTH_AND_SEVENTH_OFF(1),
+	DEMI_SEMI_QUAVERS8_SECOND_FORTH_SIXTH_AND_EIGTH_OFF(1),
+	DEMI_SEMI_QUAVERS8_SECOND_THIRD_FIFTH_AND_SIXTH_OFF(1),
+	DEMI_SEMI_QUAVERS8_FIRST_SECOND_FORTH_AND_FIFTH_OFF(1),
+	DEMI_SEMI_QUAVERS8_SECOND_FIFTH_SEVENTH_AND_EIGTH_OFF(1),
+	DEMI_SEMI_QUAVERS8_THIRD_SIXTH_AND_EIGTH_OFF(1),
+	TRIPLET_DEMI_SEMI_QUAVERS_FULL(1),
+	TRIPLET_DEMI_SEMI_QUAVERS_VAR1(1),
+	TRIPLET_DEMI_SEMI_QUAVERS_VAR2(1),
+	TRIPLET_DEMI_SEMI_QUAVERS_VAR3(1),
+	LATIN_V1(2),
+	LATIN_V2(2),
+	LATIN_V3(2),
+	LATIN_V4(2),
+	LATIN_V5(2),
+	LATIN_V6(2);
+	
+	private int beatValue;
+	Rhythmic_Value [] rhythmicValues;
+	
+	// ##########################################################
+	// #					  CONSTRUCTORS						#
+	// ##########################################################
+	private Rhythmic_Block(int beatVal){
+		this.setBeatValue(beatVal);
+		this.initialise();
+	}
+
+	// ##########################################################
+	// #					   	METHODS			   				#
+	// ##########################################################
+	// Returns beatValue field
+	public int getBeatValue() {
+		return beatValue;
+	}
+
+	// Sets the beatValue field to the value of the input parameter
+	public void setBeatValue(int beatValue) {
+		this.beatValue = beatValue;
+	}
+	
+	// Returns rhythmicValues field
+	public Rhythmic_Value[] getRhythmicValues() {
+		return rhythmicValues;
+	}
+
+	// Sets the rhythmicValues field to the value of the input parameter
+	public void setRhythmicValues(Rhythmic_Value[] rhythmicValues) {
+		this.rhythmicValues = rhythmicValues;
+	}
+	
+	// Initialises Each Rhythmic Block's rhythmicValues array with the appropriate Rhythmic Values
+	public void initialise(){
+		
+		String name = this.name();
+		if(name == "END"){
+			rhythmicValues = new Rhythmic_Value [1];
+			rhythmicValues[0] = Rhythmic_Value.MINIM_REST;
+		}
+		else if(name == "LASTNOTE"){
+			rhythmicValues = new Rhythmic_Value [1];
+			rhythmicValues[0] = Rhythmic_Value.MINIM;
+		}
+		else if(name == "MINIM1"){
+			rhythmicValues = new Rhythmic_Value [1];
+			rhythmicValues[0] = Rhythmic_Value.MINIM;
+		}
+		else if(name == "CROTCHETS1"){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.